var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -155,6 +155,31 @@ def determine_toolset(self, spec):
         # fallback to gcc if no toolset found
         return 'gcc'
 
+    def bjam_python_line(self, spec):
+        from glob import glob
+        from os.path import dirname, splitext
+        pydir = 'python%s.%s*' % spec['python'].version.version[:2]
+        incs = join_path(spec['python'].prefix.include, pydir, "pyconfig.h")
+        incs = glob(incs)
+        incs = " ".join([dirname(u) for u in incs])
+
+        pylib = 'libpython%s.%s*' % spec['python'].version.version[:2]
+        alllibs = join_path(spec['python'].prefix.lib, pylib)
+        if sys.platform == 'darwin':
+            libs = [u for u in alllibs if splitext(u)[1] == '.dylib']
+        else:
+            libs = [u for u in alllibs if splitext(u)[1] == '.so']
+
+        if len(libs) == 0:
+            libs = [u for u in alllibs if splitext(u)[1] == '.a']
+
+        libs = " ".join(libs)
+        return 'using python : %s : %s : %s : %s ;\n' % (
+            spec['python'].version,
```

Comment:
I missed that. But I'm not sure why python 3 shouldn't be allowed. Later versions of boost do work with python3. Though I can't say since when.